File: app/src/main/java/com/paradigmadigital/dagger/platform/ApplicationModule.java
package com.paradigmadigital.dagger.platform;

import android.content.Context;

import com.paradigmadigital.dagger.ui.AppCollaborator;
import com.paradigmadigital.dagger.ui.IAppCollaborator;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import eu.sesma.core.CoreCollaborator;

@Module
public class ApplicationModule {
    private final AndroidApplication application;

    public ApplicationModule(AndroidApplication application) {
        this.application = application;
    }

    @Provides
    Context provideContext() {
        return this.application;
    }

    @Provides
    @Singleton
    IAppCollaborator provideAppCollaborator() {
        return new AppCollaborator();
    }
}

File: app/src/main/java/eu/sesma/dagger/ApplicationModule.kt
package eu.sesma.dagger

import android.app.Application
import dagger.Module
import dagger.Provides

@Module
class ApplicationModule(private val application: AndroidApplication) {

    @Provides
    fun provideContext(): Application {
        return this.application
    }

//    @Provides
//    @Singleton
//    fun provideAppCollaborator(): IAppCollaborator {
//        return AppCollaborator()
//    }
}
